core/user.md

```diff
@@ -0,0 +1,258 @@
+# Nimbess User Guide
+
+## Getting Started
+
+Nimbess consists of the following runtime components:
+
+* Nimbess etcd
+* Stargazer
+* Nimbess Agent
+* BESS
+
+All of the runtime components are containerized with the exception of requiring
+the BESS kernel module to be compiled and loaded. Configuration files for
+installing each component are available for Kubernetes and will be referenced
+in this guide.
+
+## Requirements
+Nimbess leverages BESS' ability to use DPDK to create fast path networks.
+It is not required to compile DPDK on the host, however it is required to
+enable 2MB hugepages on the host. The BESS container will be deployed with
+hugepages. At a minimum you should provide at least 1024MB x N sockets on
+your host in available hugepages. It is also required to set selinux to
+permissive mode.
+
+## Compiling the BESS kernel module
+
+It is required to compile and load the BESS kernel module into the hosts where
+Nimbess will run. Execute the following as root on each server:
+
+1. git clone the BESS repo: git clone https://github.com/kot-begemot-uk/bess
+2. cd bess
+3. Follow the highlighted steps:
+<https://github.com/kot-begemot-uk/bess/blob/k8-development/build.sh#L1:L10>
+4. ./build.py kmod
+5. insmod ./core/kmod/bess.ko
+
+## Deploying Nimbess in Kubernetes
+
+Nimbess requires its own service account to be created with proper RBAC
+permissions in order to monitor and update k8s resources. Create the proper
+account by running:
+
+```bash
+kubectl apply -f https://raw.githubusercontent.com/nimbess/stargazer/master/deployments/nimbess-rbac-ns-account.yaml
+```
+Next, we need to create the Nimbess etcd, which will will be used to store
+Nimbess related information from Stargazer and the Agent:
+
+```bash
+kubectl create -f https://raw.githubusercontent.com/nimbess/stargazer/master/deployments/stargazer-etcd.yaml
+```
+Now install Stargazer:
+```bash
+kubectl create -f https://raw.githubusercontent.com/nimbess/stargazer/master/deployments/stargazer-pod.yaml
+```
+
+Finally, we need to install the Nimbess CNI binary, Agent, and start BESS. The
+following yaml file is used to deploy these, however you may need to edit
+it first in order to have the proper hugepage settings as referred to in
+[Requirements](#requirements). To deploy:
+
+```bash
+kubectl apply -f https://raw.githubusercontent.com/nimbess/nimbess-agent/master/k8s/nimbess.yaml
+```
+
```

Comment:
It'd be good to add a sample output showing the working state of everything deployed so the user knows what to expect when everything is functional.

Reply:
thanks will add